method.rb

```diff
@@ -0,0 +1,190 @@
+module Enumerable
+  def my_each
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return 'no block given' unless block_given?
+
+    output_array = []
+    i = 0
+    while i < size
+      output_array << self[i] if (yield i) == true
+      i += 1
+    end
+    output_array
+  end
+
+  def my_all?
+    return 'no block given' unless block_given?
+
+    status_cumulator = true
+    i = 0
+    loop do
+      status_cumulator &&= yield i
+      i += 1
+      break if i == size || status_cumulator == false
+    end
+    status_cumulator
+  end
+
+  def my_any?
+    return 'no block given' unless block_given?
+
+    status_cumulator = false
+    i = 0
+    loop do
+      status_cumulator ||= yield i
+      i += 1
+      break if i == size || status_cumulator == true
+    end
+    status_cumulator
+  end
+
+  def my_none?
+    return 'no block given' unless block_given?
```

Comment:
* [ ] should be a working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
#### compare the following examples by replacing the custom method with the original method in each example and rectify the error that arises.
```p %w{ant bear cat}.my_none? { |word| word.length == 5 } #=> true```
```p %w{ant bear cat}.my_none? { |word| word.length >= 4 } #=> false```
```p %w{ant bear cat}.my_none?(/d/)                        #=> true```
```p [1, 3.14, 42].my_none?(Float)                         #=> false```
```p [].my_none?                                           #=> true```
```p [nil].my_none?                                        #=> true```
```p [nil, false].my_none?                                 #=> true```
```p [nil, false, true].my_none?                           #=> false```